docs/developer.rst

```diff
@@ -192,6 +192,35 @@ Example for an application, which should reside under ``src/modules/applications
             def save(self, path, iter_count):
                 save_your_application(self.application, f"{path}/application.txt")
 
+Writing an asynchronous Module
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+A typical example for an asynchronous module is a solver which submits its job into
+the queue of some server and retrieves the result some times later. In QUARK this is
+supported via the interrupt/resume mechanism.
+
+QUARK modules may return instructions to the benchmark manager as first entry in the return value of
+pre and post-process. Currently the following instructions are supported:
+    - PROCEED
+    - INTERRUPT
+
+PROCEED: if the benchmark manger gets a PROCEED (or no instruction at all) he continues with the regular QUARK work flow.
```

Comment:
PROCEED: If the BenchmarkManager gets the instruction "PROCEED" (or no instruction at all) it continues with the regular QUARK workflow.
If the current job can be finished without getting an "INTERRUPT" instruction or an exception,
the BenchmarkManager adds "quark_job_status"=FINISHED to the metrics.